src/js/actions/export.js

```diff
@@ -67,53 +67,54 @@ define(function (require, exports) {
      * which is why we are always fetching directly from both stores prior to calling setExtensionData
      *
-     * TODO maybe this should handle a set of layers, for efficiency when called by layer actions?
-     *
      * @private
      * @param {number} documentID
-     * @param {number=} layerID Optional. If not supplied, sync doc-level metadata
+     * @param {Immutable.Iterable.<number>=} layerIDs Optional. If not supplied, sync doc-level metadata
      * @return {Promise}
      */
-    var _syncExportMetadata = function (documentID, layerID) {
+    var _syncExportMetadata = function (documentID, layerIDs) {
         var documentExports = this.flux.stores.export.getDocumentExports(documentID),
-            exportsMetadata,
-            playObject;
+            document = this.flux.stores.document.getDocument(documentID),
+            playObjects;
+
+        if (!documentExports || !document) {
+            throw new Error("Can not find Document or DocumentExports with ID " + documentID);
+        }
 
-        if (layerID || layerID === 0) {
-            var layerExports = documentExports && documentExports.getLayerExports(layerID),
+        // helper to build play object for  individual layers
+        var _buildPlayObject = function (layerID) {
+            var layerExports = documentExports.getLayerExports(layerID),
                 layerExportsArray = layerExports && layerExports.toJS() || [],
-                document = this.flux.stores.document.getDocument(documentID),
-                layer = document && document.layers.byID(layerID),
-                exportEnabled;
+                layer = document.layers.byID(layerID);
 
             if (!document || !layer) {
-                throw new Error("Could not sync export metadata for doc:" + documentID + ", layer:" + layerID);
+                throw new Error("Could not find Layer for doc:" + documentID + ", layer:" + layerID);
             }
 
-            exportEnabled = layer.exportEnabled;
-
-            exportsMetadata = {
+            var exportsMetadata = {
                 exportAssets: layerExportsArray,
-                exportEnabled: !!exportEnabled
+                exportEnabled: !!layer.exportEnabled
             };
 
             // set layer extension data
-            playObject = layerLib.setExtensionData(documentID,
+            return layerLib.setExtensionData(documentID,
                 layerID, global.EXTENSION_DATA_NAMESPACE, "exportsMetadata", exportsMetadata);
-        } else {
-            var rootExports = documentExports && documentExports.rootExports,
-            rootExportsArray = rootExports && rootExports.toJS() || [];
+        };
 
-            exportsMetadata = {
-                exportAssets: rootExportsArray
+        // prepare play objects for document or layer level, based on existence of layerIDs
+        if (layerIDs) {
+            playObjects = layerIDs.map(_buildPlayObject).toArray();
```

Comment:
At some point, we should just push the Immutable dependency into the adapter to avoid all this `toArray` nonsense.